src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/rmd/TextEditingTargetNotebook.java

```diff
@@ -1428,13 +1428,20 @@ public void execute()
    private void execAlternateEngineChunk(String docId,
                                          String chunkId,
                                          String engine,
-                                         String code)
+                                         String code,
+                                         Map<String, String> chunkOptions)
    {
+      // convert to JSO for RPC
+      JsObject jsoChunkOptions = JsObject.createJsObject();
+      for (Map.Entry<String, String> entry : chunkOptions.entrySet())
+         jsoChunkOptions.setString(entry.getKey(), entry.getValue());
+      
       server_.executeAlternateEngineChunk(
             docId,
             chunkId,
             engine,
             code,
+            jsoChunkOptions,
```

Comment:
Let's make sure that this call works when the notebook is in another source window (i.e. that the parameters all marshall correctly across a window boundary)